### PR TITLE
Support using only live cloud-transcode flavors if timeAlignedRenditions was set

### PR DESCRIPTION
**Pull request by danoss**
(PLAT-3588)

### CODE REVIEW

File: alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php [hunk 1]
```diff
@@ -794,7 +794,35 @@ private function getLiveEntryStreamConfig()
 		if(in_array($this->deliveryAttributes->getFormat(), self::$httpFormats) && !in_array($protocol, self::$httpProtocols))
 			$protocol = requestUtils::getProtocol();
 		
-		$liveStreamConfig = $this->entry->getLiveStreamConfigurationByProtocol($this->deliveryAttributes->getFormat(), $protocol, $tag, false, $this->flavorParamsIds);
+		$flavorParamsIds = $this->flavorParamsIds;
+		// use only cloud transcode flavors if timeAlignedRenditions was set
+		$partnerId = $this->entry->getPartnerId();
+		$partner = PartnerPeer::retrieveByPK($partnerId);
+		$partnerTimeAligned = $partner->getTimeAlignedRenditions();
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php]: Validate that partner is not null.

File: alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php [hunk 2]
```diff
@@ -794,7 +794,35 @@ private function getLiveEntryStreamConfig()
...
+    		    else {
+    		        $transcodeParamsIds[] = $flavor->getId();
+    		    }
+    		}
+    		 
+    		// if so, use only the transcode (if there are)
+    		if (($hasPassthrough) && count($transcodeParamsIds)) {
+    		    $flavorParamsIds = $transcodeParamsIds;
+    		}
+		}
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php]: Fix indentation

File: admin_console/forms/PartnerConfiguration.php [hunk 3]
```diff
@@ -110,6 +110,11 @@ public function init()
 				'label'	  => 'Default Entitlement Enforcement',
 				'decorators' => array('ViewHelper', array('Label', array('placement' => 'append')), array('HtmlTag',  array('tag' => 'dt', 'class' => 'partner_configuration_checkbox_field')))
 			));
+
+		$this->addElement('checkbox', 'time_aligned_renditions', array(
+		    'label'	  => 'Exclude Source from Transcoded Renditions',
```
Comment (tan-tan-kanarek) [inline admin_console/forms/PartnerConfiguration.php]: Please rename: "Kaltura Live Streams - Exclude Source Rendition"

File: alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php [hunk 4]
```diff
@@ -794,6 +794,29 @@ private function getLiveEntryStreamConfig()
...
+			{
+				// check if we have any transcode flavor
+				if (!$flavor->hasTag("ingest")) {
+					$hasTranscode = true;
+				}
+			}
+    		 
+			// if so, use only the transcode
+			if ($hasTranscode) {
+				$tag = 'mbr';
```
Comment (tan-tan-kanarek) [inline alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php]: This is wrong, there are ingested flavors with mbr tag.

**Pull request closed without merging**